src/reaper_osara.cpp

```diff
@@ -871,12 +871,50 @@ void postRenameTrack(int command) {
 	outputMessage("Track name");
 }
 
+bool isItemMuted(MediaItem* item) {
+	return *(bool*)GetSetMediaItemInfo(item, "B_MUTE", NULL);
+}
+
 void postToggleItemMute(int command) {
-	MediaItem* item = GetSelectedMediaItem(0, 0);
-	if (!item)
+	int muteCount=0;
+	int unmuteCount=0;
+	int count = CountSelectedMediaItems(0);
+	if(count==0)
+		return;
+	if(count==1)  {
+		outputMessage(isItemMuted(GetSelectedMediaItem(0,0))?"muted":"unmuted");
+		return;
+	}
+	for (int i=0; i<count; i++)
+		isItemMuted(GetSelectedMediaItem(0, i))?muteCount++:unmuteCount++;
+	ostringstream s;
+	if(muteCount>0){
+		s<<muteCount<<" item"<<((muteCount==1)?"":"s") <<" muted";
+		s<<((unmuteCount>0)?", ":"");
+	}
+	if(unmuteCount>0) 
+		s<<unmuteCount<<" item"<<((unmuteCount==1)?"":"s") <<" unmuted";
+	outputMessage(s);
+}
+
+void postToggleItemSolo(int command) {
+	bool soloed=true;
+	int itemCount=CountMediaItems(0);
+	int selectedCount = CountSelectedMediaItems(0);
+	if(selectedCount==0)
 		return;
-	outputMessage(*(bool*)GetSetMediaItemInfo(item, "B_MUTE", NULL) ?
-		"muted" : "unmuted");
+	for(int i=0; i<itemCount; i++) {
+		MediaItem* item = GetMediaItem(0, i);
+		if((!isItemSelected(item)) && (!isItemMuted(item))) 
+			soloed=false;
+	}
```

Comment:
I worry this might be slow in large projects with thousands of items. Still, I guess that might be better than no feedback. We'll have to wait and see if this is a problem in the real world.

Reply:
I tested it on a project with a ridiculous number of items, and it doesn't seam too slow.  everything else slows down about the same amount.